pkg/common/http.go

```diff
@@ -59,15 +59,18 @@ func HTTPPRequest(httpType string, uri string, jsonData []byte, authorizationTok
 
 func HTTPResponseBody(httpResponse *http.Response) ([]byte, error) {
 	// Pull out response body
-	defer httpResponse.Body.Close()
-	httpResponseBodyBytes, err := io.ReadAll(httpResponse.Body)
-	if err != nil {
-		return nil, errors.Wrapf(err, "reading http response body failed")
+	var httpResponseBodyBytes []byte
+	if httpResponse != nil && httpResponse.Body != nil {
+		defer httpResponse.Body.Close()
+		limitSize := httpResponse.ContentLength
+		const mb134 = 1 << 27 //134MB
+		if limitSize == 0 || limitSize > mb134 {
+			limitSize = mb134
+		}
+		httpResponseBodyBytes, _ = io.ReadAll(io.LimitReader(httpResponse.Body, int64(limitSize)))

```

Comment:
Why do we need to do all this instead of just reading the response body?

Reply:
Then we should make a comment that that's why we're doing this and probably put it in the description of the PR...